pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -470,8 +471,8 @@ func scaleDown(f *agonesv1.Fleet, gameServerLister listeragonesv1.GameServerList
 	nodeCounts map[string]gameservers.NodeCount, key string, isCounter bool, replicas int32,
 	aggCount, aggCapacity, minCapacity, buffer int64) (int32, bool, error) {
 	// Exit early if we're already at MinCapacity to avoid calling getSortedGameServers if unnecessary
-	if aggCapacity == minCapacity {
-		return replicas, true, nil
+	if aggCapacity <= minCapacity {
+		return max(replicas, 1), true, nil
```

Comment:
I've updated the `scaleDown` function to always maintain at least one replica. Please let me know what you think of this modification.

Reply:
```suggestion
		return max(replicas, minCapacity), true, nil
```

Should it be this @igooch ?

Reply:
I tried it on my local, `TestApplyListPolicy` passed and `TestApplyCounterPolicy/scale_down_at_MinCapacity`
failed:

 ```
/agones/pkg/fleetautoscalers/fleetautoscalers_test.go:1451
                Error:          Not equal:
                                expected: 10
                                actual  : 70
                Test:           TestApplyCounterPolicy/scale_down_at_MinCapacity
--- FAIL: TestApplyCounterPolicy/scale_down_at_MinCapacity (0.10s)
```

Reply:
Yeah, I think I missed the mark here. I _think_ you might be right, but this is making my head hurt.

I think you actually are correct in your approach here. I'd just like @igooch to confirm.

This looks right to me I think.

Reply:
Although it's unlikely for replicas < 1, the 
```
if aggCapacity <= minCapacity {
		return max(replicas, 1), true, nil
```
is correct.

I still need to take the time to run through the rest of the logic here, and make sure the tests are robust enough.

Reply:
Thank you both for your feedback. Please take your time to review everything thoroughly. I'm happy to do further adjustments to ensure everything is robust.👍🏻